shared/lib/estimators/kNeighborsRegressor.test.ts

```diff
@@ -0,0 +1,113 @@
+/**
+*  @license
+* Copyright 2021, JsData. All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* ==========================================================================
+*/
+
+import { describe, it } from 'mocha'
+import { dfd } from '../../globals'
+import { meanSquaredError } from '../metrics/metrics'
+import { Tensor1D, Tensor2D } from '@tensorflow/tfjs'
+import { KNeighborsRegressor } from './kNeighborsRegressor'
+import { assert } from 'chai'
+import { loadDiabetes } from '../datasets/datasets'
+
+type int = number
+
+// TODO: replace this with KFold as soon as its implemented
+function* kFoldIndices(
+  k: int,
+  len: int
+): IterableIterator<[Int32Array, Int32Array]> {
+  if (!(0 < k && k % 1 === 0))
+    throw new Error('kFoldIndices(k,len): k must be a positive integer.')
+  k |= 0
+
+  if (!(k <= len && len % 1 === 0))
+    throw new Error('kFoldIndices(k,len): len must be an int greater than k.')
+  len |= 0
+
+  const n = (len / k) | 0
+  let rem = len % k
+
+  const range = new Int32Array(len)
+  for (let i = len; i-- > 0; ) range[i] = i
+
```

Comment:
Why loop backward? Why have the stopping condition in the middle? And why on one line?
```
for (let i = 0; i < len; i++) {
  range[i] = i
}
```

Reply:
1. Looping backwards is on character shorter :laughing: 
  2. may have performance advantages in some fringe situation where the `list.length` or `list.size()` parameter is not propertly inlineable by the VM

Which is why I have gotten used to writing it this way. 2) does however not apply here. I will change it.